### PR TITLE
Add descriptions and salaries to Engineer and Reliability roles

### PR DESCRIPTION
**Pull request by lukemorton**

### CODE REVIEW

File: roles/as_code/lib/engineer.rb [hunk 1]
```diff
@@ -1,16 +1,45 @@
 require_relative './shared_expectations/core_engineer_expectations'
 
 JobSpec::Role.definition 'Engineer' do
-  description 'Our software delivery teams are the core of our business. They are usually comprised of a Lead Engineer, a Delivery Manager, and one or more Engineers. As an Engineer we expect someone:'
+  description <<~DESCRIPTION
+    Our engineers build software that make our customers happy, prefer problem solving over task completing, love learning from colleagues, and constantly strive to be nice humans.
```
Comment (chrisblackburn) [inline roles/as_code/lib/engineer.rb]: makeS our customers happy
may read better as: they prefer problem solving

File: roles/as_code/lib/engineer.rb [hunk 2]
```diff
@@ -1,16 +1,45 @@
...
+
+    ## What does the job entail?
+
+    We primarily write and deliver custom software to our customers. We build solutions in a variety of languages and platforms, historically Ruby on Rails, we tend to use React, JavaScript, Elixir, and Clojure these days. We will usually be responsible for setting up a customer’s infrastructure either on AWS with tools like Ansible and Terraform. Sometimes we opt for a Platform as a Service like Heroku. We will make sure there is a continuous delivery pipeline running our automated test suites before deploying to a staging environment, and a manual button for going to production.
```
Comment (chrisblackburn) [inline roles/as_code/lib/engineer.rb]: either on AWS OR...?

Comment (chrisblackburn) [reply roles/as_code/lib/engineer.rb]: Continuous Delivery

Comment (chrisblackburn) [reply roles/as_code/lib/engineer.rb]: describing the pipeline down to manual push to production seems oddly prescriptive in this context?

File: roles/as_code/lib/engineer.rb [hunk 3]
```diff
@@ -1,16 +1,45 @@
...
+
+    The software we build solves business problems. We’ve built software for karaoke, e-commerce, retail buying departments, accountants, photography studios, warehouses, supply chain companies, and ourselves. Typical projects will last 3-6 months, some customers work with us over longer periods but we like to mix up teams at around 6 months to keep things fresh.
+
+    ## What experience are we looking for?
+
+    We’ll expect you to be proficient in at least two programming languages. This is usually something like PHP/Ruby/Python/Java/C# and then JavaScript. Everyone in the organisation comes from various language backgrounds and we see this as a strength of our offering, our teams can solve problems with open minds and across many platforms. This also makes for a great learning environment.
```
Comment (chrisblackburn) [inline roles/as_code/lib/engineer.rb]: IDK if leading with PHP sets the right tone?

Comment (chrisblackburn) [reply roles/as_code/lib/engineer.rb]: (not intending to start some sort of language flamewar)

Comment (lukemorton) [reply roles/as_code/lib/engineer.rb]: lol – you just did

File: roles/as_code/lib/engineer.rb [hunk 4]
```diff
@@ -1,16 +1,45 @@
...
+
+    Code is only the half of it. The other half is communication. We expect you to be a great communicator with customers and colleagues in written and spoken English. We meet our customers face to face regularly, even working from their office, along with keeping in touch via Slack and Google Hangouts.
+
+    Everything else is optional but highly sought after. We would love you to have experience in:
+
+    - Writing code with Test Driven Development or perhaps ATDD
+    - Mentoring colleagues, providing them with guidance rather than solutions
+    - Pair programming, as we pair around 50% of the time
+    - Agile practices like Scrum, XP, and/or Kanban
```
Comment (chrisblackburn) [inline roles/as_code/lib/engineer.rb]: like should probably be such as

File: roles/as_code/lib/engineer.rb [hunk 5]
```diff
@@ -1,16 +1,45 @@
...
+    - Component based design techniques like using pattern libraries, styled components, CSS-in-JS, BEM, and/or SUIT CSS
+    - React ecosystem including a TDD approach
+    - Working with a range of SQL and NoSQL databases
+    - Infrastructure as code tools like Ansible, Chef, Puppet, and/or Saltstack
```
Comment (chrisblackburn) [inline roles/as_code/lib/engineer.rb]: As above

File: roles/as_code/lib/reliability_engineer.rb [hunk 6]
```diff
@@ -1,7 +1,47 @@
 require_relative './shared_expectations/core_engineer_expectations'
 
 JobSpec::Role.definition 'Reliability Engineer' do
-  description 'Our software delivery teams are the core of our business. They are usually comprised of a Lead Engineer, a Delivery Manager and one or more Engineers. As an Engineer we expect someone:'
+  description <<~DESCRIPTION
+    Our reliability engineers evolve and maintain applications and infrastructure for our customers. Scalability and proactive improvements are the focus and good communication skills – the necessity.
```
Comment (chrisblackburn) [inline roles/as_code/lib/reliability_engineer.rb]: role name should be capitalised

File: roles/as_code/lib/reliability_engineer.rb [hunk 7]
```diff
@@ -1,7 +1,47 @@
...
+
+    ## What does the job entail?
+
+    We primarily write and deliver custom software to our customers. We build solutions in a variety of languages and platforms, historically Ruby on Rails, we tend to use React, JavaScript, Elixir, and Clojure these days. We will usually be responsible for setting up a customer’s infrastructure either on AWS with tools like Ansible and Terraform. Sometimes we opt for a Platform as a Service like Heroku. We will make sure there is a continuous delivery pipeline running our automated test suites before deploying to a staging environment, and a manual button for going to production.
+
+    Our application and infrastructure lifetime support services are an important part of our business. Often once our delivery teams have built custom software, the applications and infrastructure will be on boarded into our lifetime support team. This team is responsible for the ongoing development of applications, adapting applications to changing requirements as our customers grow, and fixing problems as they arise. We also onboard some applications that our teams haven’t built if they pass our quality and maintainability audits.
```
Comment (chrisblackburn) [inline roles/as_code/lib/reliability_engineer.rb]: onboarded

File: roles/as_code/lib/reliability_engineer.rb [hunk 8]
```diff
@@ -1,7 +1,47 @@
...
+
+    Reliability engineers will split their time between responding to automated alerts, proactive codebase improvement, answering issues sent into our customer support desk, and ongoing feature development.
```
Comment (chrisblackburn) [inline roles/as_code/lib/reliability_engineer.rb]: issues or requests?

File: roles/as_code/lib/reliability_engineer.rb [hunk 9]
```diff
@@ -1,7 +1,47 @@
...
+
+    ## What experience are we looking for?
+
+    We’ll expect you to be a polyglot programmer versed in a mix of object and functional programming languages. We expect you to be able to jump into a project in a language, framework or configuration you’ve not worked in before and be able to fix issues, improve the codebase and build new features in it. Not only this but the right person would be adept in sharing their knowledge with others – we’d love to hear some examples of coaching and growing team members.
+
+    Multitasking, or more accurately, prioritisation is a big part of being a reliability engineer. We expect you to be able to prioritise your work, be able to switch onto resolving emergencies without being told to, and keep all customers up to date.
+
+    Code is only the half of it. The other half is communication. We expect you to be a great communicator with customers and colleagues in written and spoken English. We meet our customers face to face regularly, even working from their office, along with keeping in touch via Slack and Google Hangouts.
```
Comment (chrisblackburn) [inline roles/as_code/lib/reliability_engineer.rb]: working from their office less relevant here?

File: roles/as_code/lib/reliability_engineer.rb (hunk 7 above)
Comment (lukemorton) [reply roles/as_code/lib/reliability_engineer.rb]: Not sure about this one

**Comment by lukemorton** [comment]
@chrisblackburn Thanks – addressed.

**Pull request merged**